File: binarysearch.py
def search(l, num, low, high):
    if low > high:
        return 'not in the list'
    elif l == []:
        return 'empty list'
    else:
        mid = (high + low) / 2
        midnum = l[mid]
        if midnum == num:
            return "At index: {}".format(midnum)
        elif midnum > num:
            return search(l, num, low, mid - 1)
        else:
            return search(l, num, mid + 1, high)

File: binarysearch.erl
-module(binarysearch).
-compile(export_all).

% much was copied from rosettacode implementation

search([], Value, Low, High) -> {error, "Empty list"};

search(List, Value, Low, High) ->
    if
        Low > High -> not_found;
        true ->
            Mid = (Low + High) div 2,
            MidNum = lists:nth(Mid, List),
            if
                MidNum > Value ->
                    search(List, Value, Low, Mid - 1);
                MidNum < Value ->
                    search(List, Value, Mid + 1, High);
                true -> Mid
            end
    end.
